docs/source/tutorials/flow.md

```diff
@@ -0,0 +1,322 @@
+# An example workflow for custom parameter layout
+
+In the [Customizing your interface](../basics/customizing_interface.md) section,
+we saw how we can use the Flow argument in a LatchMetadata construct to define
+the layout and the different sections of our interface.
+
+So, let's build a workflow to show the different types of Flow elements you can use in your own workflows and
+how they're rendered in the interface.
+
+## Building the workflow
+
+We'll build a workflow for a tool that has many possible parameters-
+[fastp](https://github.com/OpenGene/fastp), an all-in-one FASTQ preprocessor,
+which should make for an interesting and comprehensive interface.
+
+First, let's define a simple dockerfile to install the tool in our environment:
+
+```Dockerfile
+# Dockerfile
+
+FROM 812206152185.dkr.ecr.us-west-2.amazonaws.com/latch-base:6839-main
+
+RUN apt-get update &&\
+    apt-get install -y curl
+
+# Install fastp
+RUN curl -L http://opengene.org/fastp/fastp -o fastp &&\
+    chmod a+x fastp
+
+# STOP HERE:
+# The following lines are needed to ensure your build environement works
+# correctly with latch.
+RUN python3 -m pip install --upgrade latch
+COPY wf /root/wf
+ARG tag
+ENV FLYTE_INTERNAL_IMAGE $tag
+WORKDIR /root
+```
+
+Now, since fastp accepts both single-end and paired-end reads as input, let's try to build this
+workflow to be able to accept both types of input - this is a great example to show how the Fork
+element can be used.
+
+First, let's work on the workflow code itself:
+
+As types for these two inputs, let's build simple dataclasses, to hold the sample names and the reads
+themselves.
+
+```py
+# wf/__init__.py
+
+import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+from dataclasses_json import dataclass_json
+from latch import medium_task, workflow
+from latch.resources.launch_plan import LaunchPlan
+from latch.types import LatchDir, LatchFile
+
+from .docs import fastp_docs
+
+
+@dataclass_json
```

Comment:
`@dataclass_json` should REALLY be deprecated
try without it and see if this works